lib/xandra.ex

```diff
@@ -206,6 +215,9 @@ defmodule Xandra do
       # Start a connection and register it under a name:
       {:ok, _conn} = Xandra.start_link(name: :xandra)
 
+      # Start a connection that requires authentication:
+      {:ok, conn} = Xandra.start_link(authentication: {MyAuthenticator, foo: "foo", bar: "bar"})
+
```

Comment:
I think we can drop this example. We're only keeping very common use cases in this "Examples" section, and we're skipping examples for stuff like clustering, compression, and so on.